llvm/test/Analysis/CtxProfAnalysis/full-cycle.ll

```diff
@@ -0,0 +1,110 @@
+; REQUIRES: x86_64-linux
+
+; RUN: split-file %s %t
+;
+; Test that the GUID metadata survives through thinlink.
+;
+; RUN: llvm-ctxprof-util fromJSON --input=%t/profile.json --output=%t/profile.ctxprofdata
+;
+; Disable pre-inline to avoid losing the trivial functions to the preinliner.
```

Comment:
This is just a limitation of the test right? If so can we avoid this by adding noinline attributes instead?

Reply:
yes. either would work

Reply:
Can you change it use the attribute, I'm a little wary of disabling critical passes we rely on in tests.